client/me/account/main.jsx

```diff
@@ -663,6 +684,85 @@ const Account = createReactClass( {
 		);
 	},
 
+	handleSubmitError( error, formName = '' ) {
+		debug( 'Error saving settings: ' + JSON.stringify( error ) );
+
+		// handle error case here
+		if ( error.message ) {
+			this.props.errorNotice( error.message, noticeOptions );
+		} else {
+			this.props.errorNotice(
+				this.props.translate( 'There was a problem saving your changes.' ),
+				noticeOptions
+			);
+		}
+
+		this.setState( {
+			submittingForm: false,
+			formsSubmitting: {
+				...this.state.formsSubmitting,
+				...( formName && { [ formName ]: false } ),
+			},
+		} );
+	},
+
+	isSubmittingForm( formName ) {
+		return formName ? this.state.formsSubmitting[ formName ] : this.state.submittingForm;
```

Comment:
Is there a chance `formName` will be falsy? Why do we need this check?

Reply:
Yes, I think for the username form `formName` will be `undefined`. This check was introduced by #48710. This form is hidden and only shown if the username text field is changed.